src/plugins/intel_cpu/src/memory_desc/dnnl_blocked_memory_desc.h

```diff
@@ -5,12 +5,15 @@
 #pragma once
 
 #include "blocked_memory_desc.h"
+#include "../../common/util/include/openvino/util/util.hpp"
 #include <cpu_memory.h>
 #include <dnnl_extension_utils.h>
 
 namespace ov {
 namespace intel_cpu {
 
+DISABLE_WARNING_MSVC_BEGIN(4250)  // Visual Studio warns us about inheritance via dominance but it's done intentionally
+                                  // so turn it off
 class DnnlBlockedMemoryDesc : public BlockedMemoryDesc, public DnnlMemoryDesc {
```

Comment:
can't we fix it using:
```c++
using BlockedMemoryDesc::method_name
```
?
With explicit selection of method we want to see in `DnnlBlockedMemoryDesc` class
I suppose compiler will not generate warnings that cannot be fixed properly, because diamond inheritance is allowed by C++ standard. Compiler just does not know which method to pick up

Reply:
@ilya-lavrenov Actually, No.
Warning occur at "DnnlBlockedMemoryDesc"'s declaration, whatever it is used or not.
In addition, based my test, I think compiler known which method to pick up. Becuase Linux have no warning, and Windows have warning, but Windows have same behaviour with Linux.

Reply:
> Warning occur at "DnnlBlockedMemoryDesc"'s declaration, whatever it is used or not.

what exact warning do we see?

Reply:
Here is simple case:
```
//DISABLE_WARNING_MSVC_BEGIN(4250)
class MemoryDesc {
public:

    virtual ~MemoryDesc() = default;

    virtual int getPrecision() const = 0;
};

class DnnlMemoryDesc : public virtual MemoryDesc {
public:

    int getPrecision() const override {
        return 1;
    }
};

class BlockedMemoryDesc : public virtual MemoryDesc {
};

class DnnlBlockedMemoryDesc : public BlockedMemoryDesc, public DnnlMemoryDesc {
    
};
//DISABLE_WARNING_MSVC_END(4250)
int main() {
    //DnnlBlockedMemoryDesc des;
    return 0;
}
```

Whatever DnnlBlockedMemoryDesc is used or not, we can all see the warnings.
```
1>C:\xiping\CMakeTestPrj\cmake_test_add_all_file_once\main.cpp(31,1): warning C4250: 'DnnlBlockedMemoryDesc': inherits 'DnnlMemoryDesc::DnnlMemoryDesc::getPrecision' via dominance
1>C:\xiping\CMakeTestPrj\cmake_test_add_all_file_once\main.cpp(21): message : see declaration of 'DnnlMemoryDesc::getPrecision'
```

Reply:
what if we use:
```c++
class DnnlBlockedMemoryDesc : public BlockedMemoryDesc, public DnnlMemoryDesc {
public:
    using DnnlMemoryDesc::getPrecision;
};
```
do we still have a warning?

Reply:
Yes. Still have warning.
`2>C:\xiping\CMakeTestPrj\cmake_test_add_all_file_once\main.cpp(31,1): warning C4250: 'DnnlBlockedMemoryDesc': inherits 'DnnlMemoryDesc::DnnlMemoryDesc::getPrecision' via dominance`